oidcserverv2/server.go

```diff
@@ -199,6 +230,107 @@ func (s *Server) Authenticate(ctx context.Context, authID string, ident oidcserv
 		return "", err
 	}
 
+	if s.consentAll ||
+		(s.consentOffline && strContains(sess.LoginRequest.Scopes, "offline_access")) {
+		return fmt.Sprintf("/consent?authid=%s", authID), nil
```

Comment:
I'm not 100% sure anyone could cause havoc here because `authID` is known to be valid by this point, but I think I would feel like this was less of a red flag if `authID` were passed through `url.QueryEscape` before being placed into a URL.

```suggestion
		return fmt.Sprintf("/consent?authid=%s", url.QueryEscape(authID)), nil
```